### PR TITLE
Persist setting sidebar status

### PR DESCRIPTION
**Pull request by na2hiro**
Hi, I thought it'd be convenient if the setting sidebar on lesson view is persisted across page lessons. Currently if a user goes back from lesson to progress or lessons page and then go to lesson again, or even refresh the page, settings will be shown again

Especially I want it to be kept hidden because I use browser's zoom to focus on what I type. When zoomed, sidebar is bigger

Screenshot 2024-04-13 at 9 33 31

Screenshot 2024-04-13 at 9 34 20

The change is to include the `hideOtherSettings` flag in `UserSettings`

### CODE REVIEW

File: src/pages/lessons/Lessons.tsx [hunk 1]
```diff
@@ -378,6 +382,7 @@ const Lessons = ({
         path={`${match.url}/:category/:subcategory/:lessonPath`}
         render={(props) => (
           <Lesson
+            key={props.location.pathname}
```
Comment (didoesdigital) [inline src/pages/lessons/Lessons.tsx:385]: What prompted adding this? I'm just curious

**Comment by didoesdigital** [comment]
Thanks for this PR @na2hiro! It looks great and this is an excellent usability improvement to preserve students' preferences across lessons 👏

File: src/pages/lessons/Lessons.tsx (hunk 1 above)
Comment (na2hiro) [reply src/pages/lessons/Lessons.tsx:385]: oops, that's a mistake. I tested something here. I'm going to remove this

**Comment by didoesdigital** [comment]
🥳 merged and deployed! Thanks, @na2hiro 👏

Note: I ran `prettier` formatting on some of the files before deploying in https://github.com/didoesdigital/typey-type/commit/a92a3b6ffde8f209c4bb74f600c0febe49bac92f. Not all files in the project have been formatted or converted to TypeScript yet so I only run Prettier on demand in some files that won't cause giant, noisy diffs. It's not important, just letting you know in case it interests you.

**Comment by na2hiro** [comment]
Thanks for quick checking and release! It's going to make my practice life smooth! And thanks for letting me know about formatting; I was wondering about it that there's not such npm script to run. Makes sense!

**Pull request merged**